src/ScottPlot4/ScottPlot/SnapLogic/NearestPosition2D.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+
+namespace ScottPlot.SnapLogic;
+
+/// <summary>
+/// Snaps to the nearest position in a user-provided array
+/// </summary>
+public class NearestPosition2D : ISnap2D
+{
+    private readonly double[] Xs;
+    private readonly double[] Ys;
+
+    public NearestPosition2D(double[] xs, double[] ys)
```

Comment:
This could maybe be an `IEnumerable<Coordinate>` instead?

Reply:
I like it! I'll make this change now